In [1]:
import pandas as pd

In [2]:
data = pd.read_excel('Med Info Inquiries 2019.xlsx')

In [3]:
data.columns

Index(['Product', 'Source', 'Category', 'Topic', 'Question', 'Response',
       'Type'],
      dtype='object')

In [4]:
data.head(5)

Product   Source             Category   Topic  \
0  ADEMPAS  2_Email  Availability/Supply   AVAIL   
1  ADEMPAS    9_PIR               Others   OTHER   
2  ADEMPAS  2_Email               Others   OTHER   
3  ADEMPAS  2_Email               Others   OTHER   
4  ADEMPAS  2_Email            Sales/Mkg  REPREQ   

                                            Question  \
0  Adempas Limited Distribution Access\n\nDear Ad...   
1                                      Illegible Fax   
2  We received a pill container with a new patien...   
3                                    See attachment.   
4                        Requesting to speak to rep.   

                                            Response                    Type  
0  3/12/19 Email sent to contact Bayer for furthe...              Pharmacist  
1  06/10/19 Email sent to Cheryl Nichols to resen...           Other Non-HCP  
2  06/18/19 Email sent to contact Bayer for furth...           Other Non-HCP  
3  07/22/19 No further action required. No return...           Other Non-HCP  
4  07/31/19 Email sent to Charlie Anderson and DM...  Other HCP (e.g. Nurse)

In [5]:
data

Product          Source             Category        Topic  \
0      ADEMPAS         2_Email  Availability/Supply        AVAIL   
1      ADEMPAS           9_PIR               Others        OTHER   
2      ADEMPAS         2_Email               Others        OTHER   
3      ADEMPAS         2_Email               Others        OTHER   
4      ADEMPAS         2_Email            Sales/Mkg       REPREQ   
5      ADEMPAS           9_PIR         Product info      GENINFO   
6      ADEMPAS       8_Website  Availability/Supply        PRICE   
7      ADEMPAS  99_IRMS-Mobile               Others  MSL REQUEST   
8      ADEMPAS  99_IRMS-Mobile               Others  MSL REQUEST   
9      ADEMPAS  99_IRMS-Mobile               Others  MSL REQUEST   
10     ADEMPAS  99_IRMS-Mobile               Others  MSL REQUEST   
11     ADEMPAS  99_IRMS-Mobile               Others  MSL REQUEST   
12     ADEMPAS  99_IRMS-Mobile               Others  MSL REQUEST   
13     ADEMPAS  99_IRMS-Mobile               Others  MSL REQUEST   
14     ADEMPAS  99_IRMS-Mobile               Others  MSL REQUEST   
15     ADEMPAS  99_IRMS-Mobile               Others  MSL REQUEST   
16     ADEMPAS  99_IRMS-Mobile                   AE      AE ONLY   
17     ADEMPAS  99_IRMS-Mobile               Others  MSL REQUEST   
18     ADEMPAS  99_IRMS-Mobile               Others  MSL REQUEST   
19     ADEMPAS  99_IRMS-Mobile               Others  MSL REQUEST   
20     ADEMPAS           9_PIR               Others  MSL REQUEST   
21     ADEMPAS  99_IRMS-Mobile               Others  MSL REQUEST   
22     ADEMPAS  99_IRMS-Mobile               Others  MSL REQUEST   
23     ADEMPAS  99_IRMS-Mobile               Others  MSL REQUEST   
24     ADEMPAS           9_PIR               Others  MSL REQUEST   
25     ADEMPAS  99_IRMS-Mobile               Others  MSL REQUEST   
26     ADEMPAS  99_IRMS-Mobile               Others  MSL REQUEST   
27     ADEMPAS  99_IRMS-Mobile               Others  MSL REQUEST   
28     ADEMPAS  99_IRMS-Mobile               Others  MSL REQUEST   
29     ADEMPAS         1_Phone               Others        OTHER   
...        ...             ...                  ...          ...   
16542   XOFIGO         1_Phone                   AE      AE ONLY   
16543   XOFIGO         1_Phone       Dose and Admin      DOSEADM   
16544   XOFIGO       8_Website             Efficacy      PRODEFF   
16545   XOFIGO           9_PIR                   AE      AE ONLY   
16546   XOFIGO           9_PIR               Safety        AEGEN   
16547   XOFIGO           9_PIR       Dose and Admin      DOSEADM   
16548   Xofigo           9_PIR        Combo therapy        COMBO   
16549   XOFIGO           9_PIR          Precautions         PREG   
16550   XOFIGO           9_PIR               Others  MSL REQUEST   
16551   XOFIGO         1_Phone       Dose and Admin       COMPAT   
16552   Xofigo           9_PIR               Others  MSL REQUEST   
16553   XOFIGO           9_PIR               Others  MSL REQUEST   
16554   XOFIGO           9_PIR               Safety        AEGEN   
16555   XOFIGO           9_PIR                   AE      AE ONLY   
16556   XOFIGO         1_Phone                   AE      AE ONLY   
16557   XOFIGO         1_Phone         Product info      GENINFO   
16558   XOFIGO         1_Phone                   AE      AE ONLY   
16559   XOFIGO         1_Phone                   AE      AE ONLY   
16560   XOFIGO         1_Phone         Product info      GENINFO   
16561   XOFIGO         1_Phone         Product info      GENINFO   
16562   XOFIGO         1_Phone        Combo therapy        COMBO   
16563   XOFIGO         6_Other               Others       NO INQ   
16564   XOFIGO     4_In Person               Others        OTHER   
16565   XOFIGO         6_Other               Others        OTHER   
16566   XOFIGO  99_IRMS-Mobile  Availability/Supply        AVAIL   
16567   XOFIGO  99_IRMS-Mobile           Indication        INDIC   
16568   XOFIGO         1_Phone                   AE      AE ONL

In [6]:
data.Product.value_counts()

MIRENA                        9320
KYLEENA                       1663
SKYLA                          903
XOFIGO                         647
VITRAKVI                       517
GADAVIST                       488
STIVARGA                       451
WHC GENERAL                    355
ADEMPAS                        348
CLIMARA PRO                    211
NEXAVAR                        197
KOGENATE FS                    195
ALIQOPA                        150
BETASERON                      140
CLIMARA                        112
NUBEQA                         109
Mirena                          75
EOVIST                          72
JIVI                            69
IMAG GENERAL                    62
ULTRAVIST                       55
KOVALTRY                        51
CIPRO                           48
DAROLUTAMIDE                    44
Xofigo                          44
ONC GENERAL                     37
Nubeqa                          35
GENERAL                         34
KOGENATE FS W VIAL A

In [7]:
data.Question.value_counts()

PTC                                                                                                                                                                                                                                                                                                                                                                                      1578
AE                                                                                                                                                                                                                                                                                                                                                                                       1307
Product Technical Complaint Form                                                                                                                                                                                                            

### Cleaning Product Names

In [8]:
def clean_product_names(x):
    return str(x).lower()

In [9]:
data['Product'] = data['Product'].apply(clean_product_names)

In [10]:
data

Product          Source             Category        Topic  \
0      adempas         2_Email  Availability/Supply        AVAIL   
1      adempas           9_PIR               Others        OTHER   
2      adempas         2_Email               Others        OTHER   
3      adempas         2_Email               Others        OTHER   
4      adempas         2_Email            Sales/Mkg       REPREQ   
5      adempas           9_PIR         Product info      GENINFO   
6      adempas       8_Website  Availability/Supply        PRICE   
7      adempas  99_IRMS-Mobile               Others  MSL REQUEST   
8      adempas  99_IRMS-Mobile               Others  MSL REQUEST   
9      adempas  99_IRMS-Mobile               Others  MSL REQUEST   
10     adempas  99_IRMS-Mobile               Others  MSL REQUEST   
11     adempas  99_IRMS-Mobile               Others  MSL REQUEST   
12     adempas  99_IRMS-Mobile               Others  MSL REQUEST   
13     adempas  99_IRMS-Mobile               Others  MSL REQUEST   
14     adempas  99_IRMS-Mobile               Others  MSL REQUEST   
15     adempas  99_IRMS-Mobile               Others  MSL REQUEST   
16     adempas  99_IRMS-Mobile                   AE      AE ONLY   
17     adempas  99_IRMS-Mobile               Others  MSL REQUEST   
18     adempas  99_IRMS-Mobile               Others  MSL REQUEST   
19     adempas  99_IRMS-Mobile               Others  MSL REQUEST   
20     adempas           9_PIR               Others  MSL REQUEST   
21     adempas  99_IRMS-Mobile               Others  MSL REQUEST   
22     adempas  99_IRMS-Mobile               Others  MSL REQUEST   
23     adempas  99_IRMS-Mobile               Others  MSL REQUEST   
24     adempas           9_PIR               Others  MSL REQUEST   
25     adempas  99_IRMS-Mobile               Others  MSL REQUEST   
26     adempas  99_IRMS-Mobile               Others  MSL REQUEST   
27     adempas  99_IRMS-Mobile               Others  MSL REQUEST   
28     adempas  99_IRMS-Mobile               Others  MSL REQUEST   
29     adempas         1_Phone               Others        OTHER   
...        ...             ...                  ...          ...   
16542   xofigo         1_Phone                   AE      AE ONLY   
16543   xofigo         1_Phone       Dose and Admin      DOSEADM   
16544   xofigo       8_Website             Efficacy      PRODEFF   
16545   xofigo           9_PIR                   AE      AE ONLY   
16546   xofigo           9_PIR               Safety        AEGEN   
16547   xofigo           9_PIR       Dose and Admin      DOSEADM   
16548   xofigo           9_PIR        Combo therapy        COMBO   
16549   xofigo           9_PIR          Precautions         PREG   
16550   xofigo           9_PIR               Others  MSL REQUEST   
16551   xofigo         1_Phone       Dose and Admin       COMPAT   
16552   xofigo           9_PIR               Others  MSL REQUEST   
16553   xofigo           9_PIR               Others  MSL REQUEST   
16554   xofigo           9_PIR               Safety        AEGEN   
16555   xofigo           9_PIR                   AE      AE ONLY   
16556   xofigo         1_Phone                   AE      AE ONLY   
16557   xofigo         1_Phone         Product info      GENINFO   
16558   xofigo         1_Phone                   AE      AE ONLY   
16559   xofigo         1_Phone                   AE      AE ONLY   
16560   xofigo         1_Phone         Product info      GENINFO   
16561   xofigo         1_Phone         Product info      GENINFO   
16562   xofigo         1_Phone        Combo therapy        COMBO   
16563   xofigo         6_Other               Others       NO INQ   
16564   xofigo     4_In Person               Others        OTHER   
16565   xofigo         6_Other               Others        OTHER   
16566   xofigo  99_IRMS-Mobile  Availability/Supply        AVAIL   
16567   xofigo  99_IRMS-Mobile           Indication        INDIC   
16568   xofigo         1_Phone                   AE      AE ONL

In [11]:
data.Product.value_counts()

mirena                        9395
kyleena                       1681
skyla                          907
xofigo                         691
vitrakvi                       537
gadavist                       488
stivarga                       469
adempas                        363
whc general                    355
climara pro                    211
nexavar                        199
kogenate fs                    195
aliqopa                        157
nubeqa                         144
betaseron                      142
climara                        112
eovist                          72
jivi                            71
imag general                    62
ultravist                       55
kovaltry                        51
cipro                           48
darolutamide                    44
onc general                     37
general                         34
kogenate fs w vial adaptor      30
larotrectinib                    9
regorafenib                      8
hem general         

### Major products
- mirena                        9395
- kyleena                       1681
- skyla                          907
- xofigo                         691
- vitrakvi                       537
- gadavist                       488
- stivarga                       469
- adempas                        363
- whc general                    355
- climara pro                    211
- nexavar                        199
- kogenate fs                    195
- aliqopa                        157
- nubeqa                         144
- betaseron                      142
- climara                        112

In [12]:
data.Topic.value_counts()

AE ONLY          3996
PTCGEN           2555
REPLACEMENT      1725
GENINFO          1503
AE and PTC       1395
OTHER             969
MSL REQUEST       842
AVAIL             578
MKTG MAT          484
SAMPLE            252
CLINDATA          236
DOSEADM           232
REPREQ            218
MSL Request       187
REIMBURSEMENT     176
AEGEN             121
F UP              104
STABIL            104
INDIC             103
PRECAUT            88
PRICE              84
PTC                77
STORE              74
FORMUL             55
CLINPART           45
PHARECO            45
PRODEFF            44
COMBO              42
MSDS               27
NAMEPT             22
PHARKIN            22
INTERDRUG          20
PRODLAB            17
PUBL               15
DISPO              14
MOA                10
SWITCH              9
CRUSH               9
MONITOR             8
COMPEFSAF           7
CONTRA              7
EPID                6
PREG                5
PROD ID             5
COMPEFF             5
RENAL     

### Epidemology Topic Questions

In [13]:
data[data['Topic'] == "EPID"]

Product          Source     Category Topic  \
423     aliqopa  99_IRMS-Mobile  Epidemology  EPID   
15225  vitrakvi  99_IRMS-Mobile  Epidemology  EPID   
15409  vitrakvi           9_PIR  Epidemology  EPID   
15433  vitrakvi           9_PIR  Epidemology  EPID   
15497  vitrakvi           9_PIR  Epidemology  EPID   
15498  vitrakvi           9_PIR  Epidemology  EPID   

                                                Question  \
423    What percentage of HCC and cholangiocarcinoma ...   
15225      What is the incidence of NTRK fusions in STS?   
15409  Please see attached PIR.is there data availabl...   
15433  Veeva ID ID-75658, Info on breakdown prevalenc...   
15497  Veeva ID ID-75880, Thanks! What is the rate in...   
15498  NTRK gene fusion frequency in melanoma patient...   

                                                Response  \
423    Chris McNabb provided the best available epide...   
15225                       O.1%. No follow-up required.   
15409  2/8/19 Escalated to oncology. (ML)\nsent an MC...   
15433  2/25/19- Escalated to Oncology Queue. (CA)\nwi...   
15497  06/10/19 Escalated to oncology. (ML)\nsent cus...   
15498  06/10/19 Paper PIR escalated to oncology. (ML)...   

                         Type  
423                 Physician  
15225               Physician  
15409               Physician  
15433  Other HCP (e.g. Nurse)  
15497               Physician  
15498               Physician

In [14]:
ae_only_df = data[data['Topic'] == "AE ONLY"]
ae_only_df

Product          Source Category    Topic  \
16     adempas  99_IRMS-Mobile       AE  AE ONLY   
30     adempas         1_Phone       AE  AE ONLY   
31     adempas         1_Phone       AE  AE ONLY   
34     adempas         1_Phone       AE  AE ONLY   
36     adempas         1_Phone       AE  AE ONLY   
47     adempas         1_Phone       AE  AE ONLY   
48     adempas         1_Phone       AE  AE ONLY   
53     adempas         1_Phone       AE  AE ONLY   
56     adempas         1_Phone       AE  AE ONLY   
59     adempas         1_Phone       AE  AE ONLY   
62     adempas         1_Phone       AE  AE ONLY   
71     adempas         1_Phone       AE  AE ONLY   
74     adempas         1_Phone       AE  AE ONLY   
76     adempas         1_Phone       AE  AE ONLY   
91     adempas         1_Phone       AE  AE ONLY   
94     adempas         1_Phone       AE  AE ONLY   
97     adempas         1_Phone       AE  AE ONLY   
103    adempas         1_Phone       AE  AE ONLY   
110    adempas         1_Phone       AE  AE ONLY   
122    adempas         1_Phone       AE  AE ONLY   
130    adempas         1_Phone       AE  AE ONLY   
131    adempas         1_Phone       AE  AE ONLY   
133    adempas         1_Phone       AE  AE ONLY   
134    adempas         1_Phone       AE  AE ONLY   
140    adempas         1_Phone       AE  AE ONLY   
143    adempas         1_Phone       AE  AE ONLY   
147    adempas         1_Phone       AE  AE ONLY   
149    adempas         1_Phone       AE  AE ONLY   
151    adempas         1_Phone       AE  AE ONLY   
167    adempas         1_Phone       AE  AE ONLY   
...        ...             ...      ...      ...   
16398   xofigo         1_Phone       AE  AE ONLY   
16408   xofigo         1_Phone       AE  AE ONLY   
16412   xofigo         1_Phone       AE  AE ONLY   
16419   xofigo         1_Phone       AE  AE ONLY   
16429   xofigo         2_Email       AE  AE ONLY   
16442   xofigo         1_Phone       AE  AE ONLY   
16452   xofigo         1_Phone       AE  AE ONLY   
16458   xofigo         1_Phone       AE  AE ONLY   
16463   xofigo         1_Phone       AE  AE ONLY   
16464   xofigo         1_Phone       AE  AE ONLY   
16474   xofigo         1_Phone       AE  AE ONLY   
16479   xofigo         1_Phone       AE  AE ONLY   
16489   xofigo         1_Phone       AE  AE ONLY   
16491   xofigo         1_Phone       AE  AE ONLY   
16507   xofigo           9_PIR       AE  AE ONLY   
16510   xofigo         1_Phone       AE  AE ONLY   
16512   xofigo         1_Phone       AE  AE ONLY   
16516   xofigo         1_Phone       AE  AE ONLY   
16524   xofigo           9_PIR       AE  AE ONLY   
16527   xofigo         1_Phone       AE  AE ONLY   
16528   xofigo           9_PIR       AE  AE ONLY   
16529   xofigo           9_PIR       AE  AE ONLY   
16540   xofigo         1_Phone       AE  AE ONLY   
16542   xofigo         1_Phone       AE  AE ONLY   
16545   xofigo           9_PIR       AE  AE ONLY   
16555   xofigo           9_PIR       AE  AE ONLY   
16556   xofigo         1_Phone       AE  AE ONLY   
16558   xofigo         1_Phone       AE  AE ONLY   
16559   xofigo         1_Phone       AE  AE ONLY   
16568   xofigo         1_Phone       AE  AE ONLY   

                                                Question  \
16                                                    AE   
30            Adverse Event Questionnaire\nPrecautionary   
31     AE Interview Form\nI am the manager calling fr...   
34                                     AE Interview Form   
36                                     AE Interview Form   
47                                                    AE   
48                           Adverse Event Questionnaire   
53                           Adverse Event Questionnaire   
56                                                    AE   
59                                                    AE   
62                           Adverse Event Questionnaire   
71     Adverse Event Questionnaire- Heidi from Adempa...   
74

In [15]:
ae_only_df['Product'].value_counts()

mirena                        2641
kyleena                        539
skyla                          181
gadavist                       156
xofigo                         153
stivarga                        55
kogenate fs                     45
adempas                         38
vitrakvi                        36
nexavar                         36
aliqopa                         16
eovist                          15
ultravist                       13
kogenate fs w vial adaptor      11
climara pro                      9
climara                          9
betaseron                        8
nubeqa                           8
cipro                            7
kovaltry                         6
whc general                      5
jivi                             5
larotrectinib                    3
onc general                      1
Name: Product, dtype: int64

In [16]:
ae_only_mirena_df = ae_only_df[ae_only_df['Product'] == 'mirena']
ae_only_mirena_df

Product   Source Category    Topic  \
3774   mirena  2_Email       AE  AE ONLY   
3778   mirena  2_Email       AE  AE ONLY   
3779   mirena  2_Email       AE  AE ONLY   
3780   mirena  2_Email       AE  AE ONLY   
3781   mirena  2_Email       AE  AE ONLY   
3782   mirena  2_Email       AE  AE ONLY   
3785   mirena  2_Email       AE  AE ONLY   
3786   mirena  2_Email       AE  AE ONLY   
3790   mirena  2_Email       AE  AE ONLY   
3791   mirena  2_Email       AE  AE ONLY   
3792   mirena  2_Email       AE  AE ONLY   
3793   mirena  2_Email       AE  AE ONLY   
3795   mirena  2_Email       AE  AE ONLY   
3798   mirena  2_Email       AE  AE ONLY   
3799   mirena  2_Email       AE  AE ONLY   
3800   mirena  2_Email       AE  AE ONLY   
3801   mirena  2_Email       AE  AE ONLY   
3802   mirena  2_Email       AE  AE ONLY   
3804   mirena  2_Email       AE  AE ONLY   
3805   mirena  2_Email       AE  AE ONLY   
3806   mirena  2_Email       AE  AE ONLY   
3807   mirena    3_Fax       AE  AE ONLY   
3808   mirena    3_Fax       AE  AE ONLY   
3813   mirena  2_Email       AE  AE ONLY   
3814   mirena  2_Email       AE  AE ONLY   
3816   mirena  2_Email       AE  AE ONLY   
3827   mirena  2_Email       AE  AE ONLY   
3828   mirena  2_Email       AE  AE ONLY   
3829   mirena  2_Email       AE  AE ONLY   
3830   mirena  2_Email       AE  AE ONLY   
...       ...      ...      ...      ...   
12814  mirena  1_Phone       AE  AE ONLY   
12825  mirena  1_Phone       AE  AE ONLY   
12827  mirena  1_Phone       AE  AE ONLY   
12830  mirena  1_Phone       AE  AE ONLY   
12832  mirena  1_Phone       AE  AE ONLY   
12870  mirena  1_Phone       AE  AE ONLY   
12910  mirena  1_Phone       AE  AE ONLY   
12912  mirena  1_Phone       AE  AE ONLY   
12919  mirena  1_Phone       AE  AE ONLY   
12925  mirena  1_Phone       AE  AE ONLY   
12953  mirena  1_Phone       AE  AE ONLY   
12966  mirena  1_Phone       AE  AE ONLY   
12977  mirena  1_Phone       AE  AE ONLY   
12983  mirena  1_Phone       AE  AE ONLY   
12986  mirena  1_Phone       AE  AE ONLY   
12994  mirena  1_Phone       AE  AE ONLY   
13001  mirena  1_Phone       AE  AE ONLY   
13012  mirena  1_Phone       AE  AE ONLY   
13017  mirena  1_Phone       AE  AE ONLY   
13028  mirena  1_Phone       AE  AE ONLY   
13029  mirena  1_Phone       AE  AE ONLY   
13063  mirena  1_Phone       AE  AE ONLY   
13071  mirena  1_Phone       AE  AE ONLY   
13094  mirena  1_Phone       AE  AE ONLY   
13104  mirena  1_Phone       AE  AE ONLY   
13106  mirena  1_Phone       AE  AE ONLY   
13128  mirena  1_Phone       AE  AE ONLY   
13137  mirena  1_Phone       AE  AE ONLY   
13142  mirena    9_PIR       AE  AE ONLY   
13157  mirena  1_Phone       AE  AE ONLY   

                                                Question  \
3774   PAR-Number=A-00052534,Status=Submitted,FollowU...   
3778   PAR-Number=A-00052741,Status=Submitted,FollowU...   
3779   PAR-Number=A-00052824,Status=Submitted,FollowU...   
3780   PAR-Number=A-00052830,Status=Submitted,FollowU...   
3781   PAR-Number=A-00052831,Status=Submitted,FollowU...   
3782   PAR-Number=A-00052838,Status=Submitted,FollowU...   
3785   PAR-Number=A-00053180,Status=Submitted,FollowU...   
3786   PAR-Number=A-00053211,Status=Submitted,FollowU...   
3790   PAR-Number=A-00053459,Status=Submitted,FollowU...   
3791   PAR-Number=A-00053935,Status=Submitted,FollowU...   
3792   PAR-Number=A-00053934,Status=Submitted,FollowU...   
3793   PAR-Number=A-00053938,Status=Submitted,FollowU...   
3795   PAR-Number=A-00054141,Status=Submitted,FollowU...   
3798   PAR-Number=A-00054431,Status=Submitted,FollowU...   
3799   PAR-Number=A-00054787,Status=Submitted,FollowU...   
3800   PAR-Number=A-00054792,Status=Submitted,FollowU...   
3801   PAR-Number=A-00054791,Status=Submitted,FollowU...   
3802   PAR-Number=A-00054872,Status=Submitted,FollowU...   
3804   PAR-Number=A-00055402,Status=Submitted,FollowU...   
3805   PAR-Number=A-00055416,Status=Submitted,FollowU...   
3806   PAR-Number

In [17]:
ae_only_mirena_df.Question.value_counts()

AE                                                                                                                                                                          462
Adverse Event Questionnaire                                                                                                                                                 120
AE Interview Form                                                                                                                                                            35
Adverse Event Questionnaire\n\nCase # provided.                                                                                                                              13
AE Update                                                                                                                                                                    13
AE update                                                                                                               

### Getting Top 10 question topics from every brand

### Major products
- mirena                        9395
- kyleena                       1681
- skyla                          907
- xofigo                         691
- vitrakvi                       537
- gadavist                       488
- stivarga                       469
- adempas                        363
- whc general                    355
- climara pro                    211
- nexavar                        199
- kogenate fs                    195
- aliqopa                        157
- nubeqa                         144
- betaseron                      142
- climara                        112

In [18]:
mirena_df = data[data['Product'] == 'mirena']
mirena_df

Product          Source         Category        Topic  \
3773   mirena       8_Website        Sales/Mkg       REPREQ   
3774   mirena         2_Email               AE      AE ONLY   
3775   mirena       8_Website        Sales/Mkg     MKTG MAT   
3776   mirena           9_PIR           Others        OTHER   
3777   mirena       8_Website        Sales/Mkg     MKTG MAT   
3778   mirena         2_Email               AE      AE ONLY   
3779   mirena         2_Email               AE      AE ONLY   
3780   mirena         2_Email               AE      AE ONLY   
3781   mirena         2_Email               AE      AE ONLY   
3782   mirena         2_Email               AE      AE ONLY   
3783   mirena         2_Email     Prod Quality       PTCGEN   
3784   mirena         2_Email           Others  REPLACEMENT   
3785   mirena         2_Email               AE      AE ONLY   
3786   mirena         2_Email               AE      AE ONLY   
3787   mirena           3_Fax           Others        OTHER   
3788   mirena           3_Fax           Others        OTHER   
3789   mirena           3_Fax           Others        OTHER   
3790   mirena         2_Email               AE      AE ONLY   
3791   mirena         2_Email               AE      AE ONLY   
3792   mirena         2_Email               AE      AE ONLY   
3793   mirena         2_Email               AE      AE ONLY   
3794   mirena       8_Website        Sales/Mkg       REPREQ   
3795   mirena         2_Email               AE      AE ONLY   
3796   mirena           3_Fax           Others        OTHER   
3797   mirena       8_Website        Sales/Mkg     MKTG MAT   
3798   mirena         2_Email               AE      AE ONLY   
3799   mirena         2_Email               AE      AE ONLY   
3800   mirena         2_Email               AE      AE ONLY   
3801   mirena         2_Email               AE      AE ONLY   
3802   mirena         2_Email               AE      AE ONLY   
...       ...             ...              ...          ...   
13138  mirena         2_Email     Product info      GENINFO   
13139  mirena         2_Email   Dose and Admin      DOSEADM   
13140  mirena           9_PIR   Dose and Admin      DOSEADM   
13141  mirena           9_PIR           Others  MSL REQUEST   
13142  mirena           9_PIR               AE      AE ONLY   
13143  mirena           9_PIR     Product info      GENINFO   
13144  mirena           9_PIR     Product info      GENINFO   
13145  mirena           9_PIR     Product info      GENINFO   
13146  mirena  99_IRMS-Mobile           Others  MSL REQUEST   
13147  mirena  99_IRMS-Mobile           Others  MSL REQUEST   
13148  mirena         1_Phone           Others        OTHER   
13149  mirena         1_Phone     Prod Quality       PTCGEN   
13150  mirena         1_Phone           Others  REPLACEMENT   
13151  mirena         1_Phone               AE   AE and PTC   
13152  mirena         1_Phone           Others  REPLACEMENT   
13153  mirena         1_Phone     Prod Quality       PTCGEN   
13154  mirena         1_Phone               AE   AE and PTC   
13155  mirena         1_Phone     Prod Quality       PTCGEN   
13156  mirena         1_Phone           Others  REPLACEMENT   
13157  mirena         1_Phone               AE      AE ONLY   
13158  mirena         1_Phone               AE   AE and PTC   
13159  mirena  99_IRMS-Mobile           Others  MSL REQUEST   
13160  mirena  99_IRMS-Mobile     Product info      GENINFO   
13161  mirena  99_IRMS-Mobile           Others  MSL REQUEST   
13162  mirena  99_IRMS-Mobile           Others  MSL REQUEST   
13163  mirena  99_IRMS-Mobile           Others  MSL REQUEST   
13164  mirena  99_IRMS-Mobile           Others  MSL REQUEST   
13165  mirena  99_IRMS-Mobile           Others  MSL REQUEST   
13166  mirena  99_IRMS-Mobile           Others  MSL REQUEST   
13167  mirena  99_IRMS-Mobile  Formul/Presentn       FORMUL   

                                                Question  \
3773   I work at a hospital that is get

In [19]:
mirena_df['Topic'].value_counts()

AE ONLY          2641
PTCGEN           1859
REPLACEMENT      1310
AE and PTC       1114
GENINFO           511
OTHER             466
AVAIL             250
MKTG MAT          231
SAMPLE            199
MSL REQUEST       189
REPREQ            101
REIMBURSEMENT      92
F UP               66
MSL Request        65
PTC                51
AEGEN              44
PRICE              43
STABIL             40
PRECAUT            28
STORE              21
FORMUL             15
PRODEFF            14
CLINDATA           11
INDIC               7
INTERDRUG           4
COMBO               3
PHARKIN             3
MSDS                3
CONTRA              2
PROD ID             2
DOSEADM             2
MOA                 1
CLINPART            1
PRODLAB             1
COMPEFF             1
PHARDYN             1
DISPO               1
FEEDBACK            1
NAMEPT              1
Name: Topic, dtype: int64

In [20]:
kyleena_df = data[data['Product'] == 'kyleena']
kyleena_df

Product          Source                    Category        Topic  \
2083  kyleena           3_Fax                          AE      AE ONLY   
2084  kyleena         2_Email                          AE      AE ONLY   
2085  kyleena       8_Website                   Sales/Mkg       REPREQ   
2086  kyleena         2_Email                          AE      AE ONLY   
2087  kyleena       8_Website                      Others        OTHER   
2088  kyleena           3_Fax                      Others        OTHER   
2089  kyleena       8_Website                   Sales/Mkg     MKTG MAT   
2090  kyleena           3_Fax                      Others        OTHER   
2091  kyleena         2_Email                          AE      AE ONLY   
2092  kyleena         2_Email                          AE      AE ONLY   
2093  kyleena         2_Email                          AE      AE ONLY   
2094  kyleena         2_Email                Prod Quality       PTCGEN   
2095  kyleena         2_Email                      Others  REPLACEMENT   
2096  kyleena           3_Fax                      Others        OTHER   
2097  kyleena         2_Email                Prod Quality       PTCGEN   
2098  kyleena         2_Email                      Others        OTHER   
2099  kyleena         2_Email                          AE   AE and PTC   
2100  kyleena         2_Email                          AE      AE ONLY   
2101  kyleena         2_Email                          AE      AE ONLY   
2102  kyleena         2_Email                          AE      AE ONLY   
2103  kyleena         2_Email                          AE      AE ONLY   
2104  kyleena         2_Email                          AE      AE ONLY   
2105  kyleena       8_Website                      Others  REPLACEMENT   
2106  kyleena         2_Email                          AE      AE ONLY   
2107  kyleena         2_Email                          AE      AE ONLY   
2108  kyleena       8_Website                   Sales/Mkg     MKTG MAT   
2109  kyleena       8_Website                   Sales/Mkg     MKTG MAT   
2110  kyleena         2_Email                          AE      AE ONLY   
2111  kyleena         2_Email                   Sales/Mkg       REPREQ   
2112  kyleena         2_Email                          AE      AE ONLY   
...       ...             ...                         ...          ...   
3734  kyleena         1_Phone                      Others  REPLACEMENT   
3735  kyleena           9_PIR                Product info      GENINFO   
3736  kyleena         1_Phone                Product info      GENINFO   
3737  kyleena           9_PIR                      Others  MSL Request   
3738  kyleena           9_PIR                Product info      GENINFO   
3739  kyleena           9_PIR                          AE      AE ONLY   
3740  kyleena         1_Phone  Storage/Stability/Disposal       STABIL   
3741  kyleena         1_Phone                          AE   AE and PTC   
3742  kyleena         1_Phone                Product info      GENINFO   
3743  kyleena         1_Phone                Prod Quality       PTCGEN   
3744  kyleena  99_IRMS-Mobile                Prod Quality       PTCGEN   
3745  kyleena  99_IRMS-Mobile                Product info      GENINFO   
3746  kyleena  99_IRMS-Mobile                          AE   AE and PTC   
3747  kyleena  99_IRMS-Mobile                Product info      GENINFO   
3748  kyleena         1_Phone                Product info      GENINFO   
3749  kyleena         1_Phone                Product info      GENINFO   
3750  kyleena         1_Phone                Prod Quality       PTCGEN   
3751  kyleena         1_Phone                          AE   AE and PTC   
3752  kyleena           9_PIR                Product info      GENINFO   
3753  kyleena           9_PIR                          AE      AE ONLY   
3754  kyleena           9_PIR                      Others  MSL REQUEST   
3755  kyleena  99_IRMS-Mobile                      Others  MSL REQUEST   
3756  kyle

In [21]:
kyleena_df['Topic'].value_counts()

AE ONLY          539
PTCGEN           306
GENINFO          188
REPLACEMENT      180
AE and PTC       131
OTHER             62
MSL REQUEST       57
MKTG MAT          44
AVAIL             27
SAMPLE            20
REPREQ            19
REIMBURSEMENT     19
MSL Request       17
PRECAUT           13
PRODLAB           11
PTC                8
AEGEN              7
STABIL             5
DOSEADM            5
PRODEFF            4
F UP               4
PRICE              3
CLINDATA           3
STORE              3
INDIC              2
DISPO              1
CLINPART           1
GENDER             1
PREG               1
Name: Topic, dtype: int64

In [22]:
skyla_df = data[data['Product'] == 'skyla']
skyla_df

Product          Source                    Category          Topic  \
13557   skyla           3_Fax                      Others          OTHER   
13558   skyla       8_Website                   Sales/Mkg         REPREQ   
13559   skyla           3_Fax                      Others          OTHER   
13560   skyla       8_Website                   Sales/Mkg         REPREQ   
13561   skyla       8_Website                   Sales/Mkg       MKTG MAT   
13562   skyla         2_Email                          AE        AE ONLY   
13563   skyla         2_Email                          AE        AE ONLY   
13564   skyla         2_Email                          AE        AE ONLY   
13565   skyla         2_Email                Prod Quality         PTCGEN   
13566   skyla         2_Email                      Others    REPLACEMENT   
13567   skyla         2_Email                      Others    REPLACEMENT   
13568   skyla         2_Email                Prod Quality         PTCGEN   
13569   skyla         2_Email                      Others    REPLACEMENT   
13570   skyla         2_Email                          AE        AE ONLY   
13571   skyla           3_Fax                      Others          OTHER   
13572   skyla           3_Fax                      Others          OTHER   
13573   skyla           3_Fax                      Others          OTHER   
13574   skyla           3_Fax                      Others          OTHER   
13575   skyla         1_Phone                      Others    REPLACEMENT   
13576   skyla         1_Phone                Prod Quality         PTCGEN   
13577   skyla         1_Phone                      Others  REIMBURSEMENT   
13578   skyla         1_Phone                          AE     AE and PTC   
13579   skyla         1_Phone                          AE     AE and PTC   
13580   skyla         1_Phone                Prod Quality         PTCGEN   
13581   skyla         1_Phone                      Others  REIMBURSEMENT   
13582   skyla         1_Phone                Prod Quality         PTCGEN   
13583   skyla         1_Phone                          AE        AE ONLY   
13584   skyla         1_Phone                          AE     AE and PTC   
13585   skyla         1_Phone                          AE     AE and PTC   
13586   skyla         1_Phone                Prod Quality         PTCGEN   
...       ...             ...                         ...            ...   
14434   skyla         1_Phone  Storage/Stability/Disposal         STABIL   
14435   skyla         1_Phone                      Others    REPLACEMENT   
14436   skyla         1_Phone                Prod Quality         PTCGEN   
14437   skyla         1_Phone                Product info        GENINFO   
14438   skyla         1_Phone                   Sales/Mkg       MKTG MAT   
14439   skyla         1_Phone         Availability/Supply          AVAIL   
14440   skyla         1_Phone  Storage/Stability/Disposal          DISPO   
14441   skyla         1_Phone                      Others  REIMBURSEMENT   
14442   skyla         1_Phone         Availability/Supply          AVAIL   
14443   skyla         1_Phone         Availability/Supply          AVAIL   
14444   skyla         1_Phone                      Others  REIMBURSEMENT   
14445   skyla         1_Phone                   Sales/Mkg       MKTG MAT   
14446   skyla         1_Phone                      Others          OTHER   
14447   skyla         1_Phone                   Sales/Mkg       MKTG MAT   
14448   skyla         1_Phone                          AE        AE ONLY   
14449   skyla         1_Phone         Availability/Supply          AVAIL   
14450   skyla         1_Phone                Prod Quality         PTCGEN   
14451   skyla         1_Phone                      Others    REPLACEMENT   
14452   skyla         1_Phone                   Sales/Mkg       MKTG MAT   
14453   skyla         1_Phone                   Sales/Mkg       MKTG MAT   
14454   skyla         1_Phone              Dose and Ad

In [23]:
skyla_df.Topic.value_counts()

AE ONLY          181
PTCGEN           179
REPLACEMENT      113
GENINFO          107
AE and PTC        73
AVAIL             49
MKTG MAT          44
OTHER             34
REIMBURSEMENT     30
STABIL            24
SAMPLE            12
F UP              10
STORE              8
REPREQ             8
PRICE              8
MSL REQUEST        7
DISPO              6
PTC                4
AEGEN              2
DOSEADM            2
DISCONT            1
CONTRA             1
MSL Request        1
FORMUL             1
PRECAUT            1
PRODLAB            1
Name: Topic, dtype: int64

In [24]:
skyla_df = data[data['Product'] == 'skyla']
skyla_df.Topic.value_counts()

AE ONLY          181
PTCGEN           179
REPLACEMENT      113
GENINFO          107
AE and PTC        73
AVAIL             49
MKTG MAT          44
OTHER             34
REIMBURSEMENT     30
STABIL            24
SAMPLE            12
F UP              10
STORE              8
REPREQ             8
PRICE              8
MSL REQUEST        7
DISPO              6
PTC                4
AEGEN              2
DOSEADM            2
DISCONT            1
CONTRA             1
MSL Request        1
FORMUL             1
PRECAUT            1
PRODLAB            1
Name: Topic, dtype: int64

In [26]:
def getTopTopics(product):
    df_name = f"{product}_df"
    df_name = data[data['Product'] == f"{product}"]
    print(f"Top Question Topics for {product} brand. \n")
    print(df_name.Topic.value_counts())
    print("\n\n")

In [27]:
products = list(set(data['Product'].values))
print(products)

['mirena', 'imag general', 'betaseron', 'darolutamide', 'regorafenib', 'xofigo', 'adempas', 'gadavist', 'climara', 'onc general', 'whc general', 'nubeqa', 'skyla', 'ultravist', 'climara pro', 'stivarga', 'nan', 'vitrakvi', 'larotrectinib', 'aliqopa', 'hem general', 'jivi', 'cipro', 'kovaltry', 'general', 'neuro general', 'nexavar', 'eovist', 'kyleena', 'kogenate fs w vial adaptor', 'kogenate fs']


In [28]:
for product in products:
    df_name = f"{product}_df"
    print(f"Brand Name = {product}\n")
    getTopTopics(product)

Brand Name = mirena

Top Question Topics for mirena brand. 

AE ONLY          2641
PTCGEN           1859
REPLACEMENT      1310
AE and PTC       1114
GENINFO           511
OTHER             466
AVAIL             250
MKTG MAT          231
SAMPLE            199
MSL REQUEST       189
REPREQ            101
REIMBURSEMENT      92
F UP               66
MSL Request        65
PTC                51
AEGEN              44
PRICE              43
STABIL             40
PRECAUT            28
STORE              21
FORMUL             15
PRODEFF            14
CLINDATA           11
INDIC               7
INTERDRUG           4
COMBO               3
PHARKIN             3
MSDS                3
CONTRA              2
PROD ID             2
DOSEADM             2
MOA                 1
CLINPART            1
PRODLAB             1
COMPEFF             1
PHARDYN             1
DISPO               1
FEEDBACK            1
NAMEPT              1
Name: Topic, dtype: int64



Brand Name = imag general

Top Question Topics for i